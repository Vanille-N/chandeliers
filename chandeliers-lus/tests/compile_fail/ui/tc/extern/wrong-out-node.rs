//! Node exists but has a different signature.
use chandeliers_sem::traits::*;
use chandeliers_sem::*;

#[allow(non_camel_case_types)]
#[derive(Debug, Default)]
struct wrong_step_sig {}
impl Step for wrong_step_sig {
    type Input = (i64, i64);
    type Output = f64;
    fn step(&mut self, _: (ty!(i64), ty!(i64))) -> ty!(f64) {
        unimplemented!()
    }
}
chandeliers_lus::decl! {
    extern node wrong_step_sig(i, j : int) returns (b: bool);
}

fn main() {}
