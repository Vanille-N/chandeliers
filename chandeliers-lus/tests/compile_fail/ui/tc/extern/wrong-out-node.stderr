error[E0308]: mismatched types
  --> tests/compile_fail/ui/tc/extern/wrong-out-node.rs:16:17
   |
16 |     extern node wrong_step_sig(i, j : int) returns (b: bool);
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^--------
   |                 |                                   |
   |                 |                                   expected `Nillable<bool>` because of return type
   |                 expected `Nillable<bool>`, found `Nillable<f64>`
   |
   = note: expected enum `Nillable<bool>`
              found enum `Nillable<f64>`
